### PR TITLE
 Add change to read certificate without restarting the server.

### PR DESCRIPTION
**Pull request by sinexu**

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -68,15 +70,87 @@ func main() {
...
+		return nil, err
+	}
+	result := &certificateReloader{
+		certPath: certPath,
+		keyPath:  keyPath,
+	}
+	result.cert = &cert
+
+	// TODO: switch to some other time interval?
+	tickerChannel := time.NewTicker(time.Minute * 30).C
```
Comment (ohfunc) [inline server/server.go]: Would it make sense to provide the interval as a flag, or at least set it as a constant at the top of the file?

File: server/server.go [hunk 2]
```diff
@@ -68,15 +70,87 @@ func main() {
...
+func (cr *certificateReloader) maybeReload() error {
+	newCert, err := tls.LoadX509KeyPair(cr.certPath, cr.keyPath)
+	if err != nil {
+		return err
+	}
+	cr.certMu.Lock()
+	defer cr.certMu.Unlock()
+	cr.cert = &newCert
+
+	log.Printf("Reloading certificate successfully!")
```
Comment (ohfunc) [inline server/server.go]: nit: s/reloading/reloaded

File: server/server.go [hunk 3]
```diff
@@ -68,15 +70,87 @@ func main() {
...
+			GetCertificate: cr.GetCertificateFunc(),
+		},
+		Handler: mux,
+	}
+	log.Fatal(server.ListenAndServeTLS("", ""))
 }
 
+///////////////////////////////////////////////
+type certificateReloader struct {
+	certMu   sync.RWMutex
```
Comment (ohfunc) [inline server/server.go]: Would it make sense to embed the mutex here?

Comment (sinexu) [reply server/server.go]: Thanks!

**Review by ohfunc — APPROVED** [review]
LGTM for Go readability

**Review by akyyy — APPROVED** [review]

**Pull request merged**